### ISSUE TITLE
Reproduce Bert4Rec's results on Beauty, and provide better SASRec results by improving the loss function.

### ISSUE THREAD

**Issue by Li-fAngyU** [asash/bert4rec_repro#5]
@asash First, thank you for your great job on the sequential recommendation.

I have noticed that in your <A Systematic Review and Replicability Study of BERT4Rec for Sequential Recommendation> article, the results of BERT4Rec on the Beauty dataset cannot be replicated. After investigation, I found that it may be due to inconsistent preprocessing methods on the dataset. I used the Beauty dataset processed by [S3Rec](https://github.com/aHuiWang/CIKM2020-S3Rec) (which can replicate the results of SASRec on Beauty) for the experiment. Finally, the experimental results of BERT4Rec can be reproduced.

Inspired by you, I am trying to improve SASRec. As you said in the paper, the main difference between BERT4Rec and SASRec lies in the training objectives or loss function. So I tried to improve the loss function of SASRec. Finally, SASRec using the improved loss function can surpass BERT4Rec on the ML-1M and Beauty datasets, and achieve similar results on the ML-20M and Steam datasets.

I provide [code](https://github.com/Li-fAngyU/bert4rec_repro) (by fork) to reproduce what I said above. Finally, I hope my findings can be helpful to you.

Thank you again for your great work!
